<?xml version="1.0"?>
<launch>
  <arg name="namespace" default="flir_boson"/>
  <arg name="frame_id" default="boson_camera" />
  <arg name="manager" default="$(arg frame_id)_nodelet_manager" />
  
  <!-- the linux file descriptor location for the camera -->
  <arg name="dev" default="/dev/v4l/by-id/usb-FLIR_Boson_103087-video-index0"/>

  <!-- valid values are 30.0 or 60.0 for Bosons -->
  <arg name="frame_rate" default="10.0"/>

  <!-- valid values are RAW16 or YUV -->
  <arg name="video_mode" default="RAW16"/>

  <!-- valid values are TRUE or FALSE -->
  <arg name="zoom_enable" default="FALSE"/>

  <!-- valid values are Boson_320 or Boson_640 -->
  <arg name="sensor_type" default="Boson_320"/>

  <!-- valid values are TRUE or FALSE -->
  <arg name="color_lut_enable" default="FALSE"/>

  <!-- valid values are TRUE or FALSE -->
  <arg name="publish_8bit_enable" default="FALSE"/>

  <!-- location of the camera calibration file -->
  <arg name="camera_info_url" default="package://flir_adk_ros/example_calibrations/"/>
  <param name="lib_path" type="str" value=""/>
  <param name="cam_count" type="int" value="1"/>

  <param name="serial_device" type="str" value="/dev/ttyACM"/>
  <node pkg="flir_adk_ros" name="info_server" type="info_server.py"/>
  <node pkg="flir_adk_ros" name="ffc_server" type="ffc_server.py"/>
  <node pkg="flir_adk_ros" name="reboot_server" type="reboot_server.py"/>
  <node pkg="flir_adk_ros" name="sync_server" type="sync_server.py"/>
  <node pkg="flir_adk_ros" name="initialize_thermal" type="initialize_thermal.py" output="screen"/>


    <!-- start nodelet manager -->
    <node pkg="nodelet" type="nodelet" name="$(arg manager)" args="manager" />

    <node pkg="nodelet" type="nodelet" name="$(arg manager)_driver" args="load flir_boson_usb/BosonCamera $(arg manager)">
      <param name="frame_id" type="str" value="$(arg frame_id)"/>
      <param name="dev" type="str" value="$(arg dev)"/>
      <param name="frame_rate" type="double" value="$(arg frame_rate)"/>
      <param name="video_mode" type="str" value="$(arg video_mode)"/>
      <param name="zoom_enable" type="bool" value="$(arg zoom_enable)"/>
      <param name="color_lut_enable" type="bool" value="$(arg color_lut_enable)"/>
      <param name="publish_8bit_enable" type="bool" value="$(arg publish_8bit_enable)"/>
      <param name="sensor_type" type="str" value="$(arg sensor_type)"/>
      <param name="camera_info_url" type="str" value="$(arg camera_info_url)Boson320.yaml"/>
    </node>
</launch>
